Source/Adobe.Target.Client/OnDevice/DecisioningEvaluator.cs

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+namespace Adobe.Target.Client.OnDevice
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Adobe.Target.Client.Model;
+    using Adobe.Target.Client.Model.OnDevice;
+    using Adobe.Target.Client.Util;
+
+    internal sealed class DecisioningEvaluator
+    {
+        private const string RemoteActivities = "Remote activities in: ";
+        private const string Mboxes = "mboxes ";
+        private const string Views = "views ";
+
+        private readonly RuleLoader ruleLoader;
+
+        public DecisioningEvaluator(RuleLoader ruleLoader) => this.ruleLoader = ruleLoader;
+
+        internal OnDeviceDecisioningEvaluation EvaluateLocalExecution(TargetDeliveryRequest request)
+        {
+            if (request == null)
+            {
+                return new OnDeviceDecisioningEvaluation(false, Messages.RequestNonNull);
+            }
+
+            var ruleSet = this.ruleLoader.GetLatestRules();
+            if (ruleSet == null)
+            {
+                return new OnDeviceDecisioningEvaluation(false, Messages.RulesetUnavailable);
+            }
+
+            var remoteMboxes = this.ComputeRemoteMboxes(request, ruleSet);
+            var remoteViews = this.ComputeRemoteViews(request, ruleSet);
```

Comment:
Here it should probably be `hasRemoteViews` and `hasRemoteMboxes` since the artifact is single.

Reply:
Fixed